### PR TITLE
[NADE] Add feature check to allow codegen

### PR DESCRIPTION
**Pull request by sfc-gh-bgoel**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added or updated automated unit tests to verify correctness of my new code - N/A
   * [ ] I've added or updated integration tests to verify correctness of my new code - N/A
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually - N/A
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes - N/A
   * [x] I've described my changes in the section below.

### Changes description
Add a feature check in `build_bundle` to allow for codegen.

### CODE REVIEW

File: src/snowflake/cli/plugins/nativeapp/manager.py [hunk 1]
```diff
@@ -47,6 +50,7 @@
     MissingPackageScriptError,
     UnexpectedOwnerError,
 )
+from snowflake.cli.plugins.nativeapp.feature_flags import FeatureFlag
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/nativeapp/manager.py:53]: I know this is not the gist of the change but should we consider renaming `FeatureFlag` to `NativeAppFeatureFlags`? In this way we may avoid any ambiguity with existing `snowflake.cli.api.feature_flags.FeatureFlag.` WDYT?

Comment (sfc-gh-bdufour) [reply src/snowflake/cli/plugins/nativeapp/manager.py:53]: Isn't it common already to have the same files in multiple plugin directories if they define the same concepts? I'm thinking commands and manager, for example. Native apps code shouldn't rely on feature flags outside of its own IMHO, so at least now there shouldn't be ambiguity.

**Comment by sfc-gh-bdufour** [comment]
Bhumika, not following the target branch here. Shouldn't we target feature-snowpark-annotations instead? Let's not create complex branching/merging structures, it will be a pain to deal with later. This should stack on top of your other work, on the common feature branch.

File: src/snowflake/cli/plugins/nativeapp/manager.py [hunk 2]
```diff
@@ -309,7 +313,14 @@ def build_bundle(self) -> ArtifactDeploymentMap:
         """
         Populates the local deploy root from artifact sources.
         """
-        return build_bundle(self.project_root, self.deploy_root, self.artifacts)
+        mapped_files = build_bundle(self.project_root, self.deploy_root, self.artifacts)
+        if FeatureFlag.ENABLE_SETUP_SCRIPT_GENERATION.is_enabled():
+            _find_and_execute_processors(
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/plugins/nativeapp/manager.py:318]: remind me: where does the output of this step go?

Comment (sfc-gh-bgoel) [reply src/snowflake/cli/plugins/nativeapp/manager.py:318]: We do not have the generate DDL piece in yet, which I can direct to stdout. 
If you just want to see the JSON output for now, I can output that instead. Will add the change in the next commit.

**Review by sfc-gh-bgoel — COMMENTED** [review]
Bruno, yes you are right that the target branch needs to be the feature branch I created. But this should either be 1. Automatically changed by github once I merge bgoel-collect-external-functions or 2. I retarget it to feature-snowflake-snowpark.

The current way of targeting was to isolate changes from just this enablement.

**Review by sfc-gh-bdufour — APPROVED** [review]

**Pull request merged**